src/Core/Utils/SystemUtils.hpp

```
@@ -93,4 +101,59 @@ namespace SystemUtils {
+            UNSUPPORTED,
+            CANNOT_BE_LOCATED,
+            CANNOT_BE_LOADED
+        };
+
+        bool present = false;
+        ErrorType errType = NONE;
+    };
+    /* Checks if the Vulkan loader exists on the current system.
+        @return The operation's result and metadata.
+    */
+    inline VkLoaderDiag VulkanLoaderExists() {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)